src/main/java/duke/task/TaskWithDateTime.java

```diff
@@ -0,0 +1,90 @@
+package duke.task;
+
+import java.time.DateTimeException;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Represents a task with an associated date and/or time.
+ *
+ * Note: the allowable formats for date and time to be interpreted correctly by TaskWithDateTime
+ * is YYYY/MM/DD (with ./| being valid separators) and HHMM.
+ *
+ * @author  Elumalai Oviya Dharshini
+ * @version 0.1
+ */
+public class TaskWithDateTime extends Task {
```

Comment:
I like how you make another TaskWithDateTime class that differentiate between normal task and task with date and time!